scale/source/messages/purge_source_file.py

```diff
@@ -0,0 +1,98 @@
+"""Defines a command message that purges a source file"""
+from __future__ import unicode_literals
+
+import logging
+
+from ingest.models import Ingest
+from job.models import JobInputFile
+from messaging.messages.message import CommandMessage
+from recipe.models import RecipeInputFile
+from storage.models import ScaleFile
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_purge_source_file_message(source_file_id, trigger_id, purge):
+    """Creates messages to removes a source file form Scale
+
+    :param source_file_id: The source file ID
+    :type source_file_id: int
+    :param trigger_id: The trigger event ID for the purge operation
+    :type trigger_id: int
+    :param purge: Boolean value to determine if files should be purged from workspace
+    :type purge: bool
+    :return: The purge source file message
+    :rtype: :class:`storage.messages.purge_source_file.PurgeSourceFile`
+    """
+
+    message = PurgeSourceFile()
+    message.source_file_id = source_file_id
+    message.trigger_id = trigger_id
+    message.purge = purge
+
+    return message
+
+
+class PurgeSourceFile(CommandMessage):
+    """Command message that removes source file models
+    """
+
+    def __init__(self):
+        """Constructor
+        """
+
+        super(PurgeSourceFile, self).__init__('purge_source_file')
+
+        self.source_file_id = None
+        self.trigger_id = None
+        self.purge = False
```

Comment:
Again I don't think you need the purge flag since the point of the message is to do a purge.